source/NationalInstruments/jquery.flot.historybuffer.analogWaveform.js

```diff
@@ -158,20 +169,38 @@ console.log(hb1.toDataSeries()); //[[4, 1], [5, 2], [6, 3], [null, null], [1, 1]
         var result = [];
         var waveforms = this.buffers[index].toArray();
         var previousWaveform = waveforms[0];
+        var beforeSkippedWaveformIndex = -1;
+        var afterSkippedWaveformIndex = -1;
+        var i = -1;
 
         waveforms.forEach(function (waveform) {
-            if (!waveformInRange(waveform, start, end)) {
-                return;
+            i++;
+            var waveformVisibility = waveformInRange(waveform, start, end);
+            if (waveformVisibility !== 0) {
+                if (waveformVisibility > 0 && afterSkippedWaveformIndex < 0) {
+                    afterSkippedWaveformIndex = i; // do not early return
```

Comment:
Since we don't return here, do we always append the first waveform after the given range to the result buffer? Is that the behavior we want?

Reply:
> Since we don't return here, do we always append the first waveform after the given range to the result buffer? Is that the behavior we want?

No, because in this case, we will check the length of 'result' before appending the extra sample. If it is empty, we do nothing.